### ISSUE TITLE
Detect possible simplifications using walrus

### ISSUE THREAD

**Issue by sam-s** [pylint-dev/pylint#9532]
### Current problem

Here is the code that should be flagged by a walrus checker:
```
foo = ....
if foo:
    ....
```
the better code is
```
if foo := ...:
    ...
```

### Desired solution

there should be a "code review" checker that flags code where 
* a variable is set and 
* its value is immediately checked and 
* is is not used if the check fails

**Comment by Pierre-Sassoulas** [comment]
Sound like https://pylint.readthedocs.io/en/stable/user_guide/messages/refactor/consider-using-assignment-expr.html, you need to activate the code style checker:
```ini
[MAIN]
py-version=3.8
load-plugins=pylint.extensions.code_style
```
